db/types/base.py

```diff
@@ -1,6 +1,72 @@
+from enum import Enum
+
 from sqlalchemy import create_engine
+
 from db import constants
 
+
+class PostgresType(Enum):
+    """
+    This only includes built-in Postgres types that SQLAlchemy supports.
+    SQLAlchemy doesn't support XML. See zzzeek's comment on:
+    https://stackoverflow.com/questions/16153512/using-postgresql-xml-data-type-with-sqlalchemy
+    The values are keys returned by get_installed_types.
+    """
+    _ARRAY = '_array'
+    BIGINT = 'bigint'
+    BIT_VARYING = 'bit varying'
+    BIT = 'bit'
+    BOOLEAN = 'boolean'
+    BYTEA = 'bytea'
+    CHAR = '"char"'
+    CHARACTER_VARYING = 'character varying'
+    CHARACTER = 'character'
+    CIDR = 'cidr'
+    DATE = 'date'
+    DATERANGE = 'daterange'
+    DECIMAL = 'decimal'
+    DOUBLE_PRECISION = 'double precision'
+    FLOAT = 'float'
+    HSTORE = 'hstore'
+    INET = 'inet'
+    INT4RANGE = 'int4range'
+    INT8RANGE = 'int8range'
+    INTEGER = 'integer'
+    INTERVAL = 'interval'
+    JSON = 'json'
+    JSONB = 'jsonb'
+    MACADDR = 'macaddr'
+    MONEY = 'money'
+    NAME = 'name'
+    NUMERIC = 'numeric'
+    NUMRANGE = 'numrange'
+    OID = 'oid'
+    REAL = 'real'
+    REGCLASS = 'regclass'
+    SMALLINT = 'smallint'
+    TEXT = 'text'
+    TIME = 'time'
+    TIME_WITH_TIME_ZONE = 'time with time zone'
+    TIME_WITHOUT_TIME_ZONE = 'time without time zone'
+    TIMESTAMP = 'timestamp'
+    TIMESTAMP_WITH_TIMESTAMP_ZONE = 'timestamp with time zone'
+    TIMESTAMP_WITHOUT_TIMESTAMP_ZONE = 'timestamp without time zone'
+    TSRANGE = 'tsrange'
+    TSTZRANGE = 'tstzrange'
+    TSVECTOR = 'tsvector'
+    UUID = 'uuid'
+
+
+class MathesarCustomType(Enum):
+    """
+    This is a list of custom Mathesar DB types.
+    Keys returned by get_installed_types are of the format 'mathesar_types.VALUE'
+    """
+    EMAIL = 'email'
+    URI = 'uri'
+    MONEY = 'money'
```

Comment:
This results in these strings being defined both here, and in the individual types.  How about using this definition in the `db.types.email` module?  Then creating a new custom type would use this, and anywhere that type is used, the name could be updated by updating the type name here (along with the type schema, if desired).

Reply:
I updated the `db.types.email` module to use the new enum.